### PR TITLE
Change run_in_executor to return a Future rather than be a coroutine

### PR DESCRIPTION
**Pull request by Caligatio**
Here's a code snippet that currently causes a `mypy` error:
```python
loop = asyncio.get_running_loop()
future = loop.run_in_executor(
    previously_defined_threadpool_executor,
    my_sync_function_that_returns_a_custom_class
)

print(type(future))
future.add_done_callback(future_exception_catcher)
```

Running the code works as expected and returns
```console
<class '_asyncio.Future'>
```

Running mypy, however, returns an error:
```console
error: "Coroutine[Any, Any, CustomClass]" has no attribute "add_done_callback"
```

[Per the docs](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) (and usage), `run_in_executor` returns a `Future` and is not a coroutine/return a coroutine.

### CODE REVIEW

File: stdlib/3/asyncio/events.pyi [hunk 1]
```diff
@@ -100,8 +100,8 @@ class AbstractEventLoop(metaclass=ABCMeta):
     @abstractmethod
     def call_soon_threadsafe(self, callback: Callable[..., Any], *args: Any) -> Handle: ...
     @abstractmethod
-    async def run_in_executor(self, executor: Any,
-                              func: Callable[..., _T], *args: Any) -> _T: ...
+    def run_in_executor(self, executor: Any,
```
Comment (hauntsaninja) [inline stdlib/3/asyncio/events.pyi:103]: AbstractEventLoop still has an async def for `run_in_executor`. But maybe typeshed's tests complain if you keep it that way.

Comment (Caligatio) [reply stdlib/3/asyncio/events.pyi]: Good catch!  Reverting the change on the AbstractEventLoop and seeing how the tests like it.

Comment (Caligatio) [reply stdlib/3/asyncio/events.pyi]: Yea, mypy does not like the change in signature:

```console
error: Return type "Future[_T]" of "run_in_executor" incompatible with return type "Coroutine[Any, Any, _T]" in supertype "AbstractEventLoop"
```

I don't want to misdocument the base class; should I put a `# type: ignore` on my `BaseEventLoop` changes?  Can you even do that in .pyi files?

Comment (Caligatio) [reply stdlib/3/asyncio/events.pyi]: Reverted the change to `AbstractEventLoop` and instead added an ignore statement with explanation.

**Comment by Caligatio** [comment]
Just to promote the resolution out of the code review bit above:

- `AbstractEventLoop` in `stdlib/3/asyncio/events.pyi` (and in implementation) declare `run_in_executor` as a coroutine `@abstractmethod`.
- Currently, `BaseEventLoop` in `stdlib/3/asyncio/base_events.pyi` has a "concrete" implementation of `run_in_executor` also as a coroutine.
- In real implementation, `BaseEventLoop.run_in_executor` returns a `Future`.

mypy understandably does not a concrete class changing the signature of its parent abstract class but that's what is in the actual implementation.  My solution is to correct the concrete implementation by declaring it as returning a `Future`  and include a `types: ignore` with a justification comment.

**Pull request merged**